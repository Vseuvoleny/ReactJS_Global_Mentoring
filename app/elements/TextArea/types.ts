export interface ITextArea {
    title: string;
    value: string;
    onChange: Function;
    errors: any;
    touched: any;
}