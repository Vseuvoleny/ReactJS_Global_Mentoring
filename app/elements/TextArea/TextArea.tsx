import React from "react";
import PropTypes from "prop-types";
import classes from "./TextArea.module.scss";

const TextArea = ({ title, value, onChange, ...rest }) => {
  const { errors, touched } = rest;
  return (
    <div className={classes.textarea_container}>
      <label html={title} className={classes.textarea_container__title}>
        {title}
      </label>
      <textarea
        id={title}
        className={classes.textarea_container__textarea}
        placeholder="Movie description"
        value={value}
        onChange={onChange}
      ></textarea>
      {errors && touched.overview && (
        <div className={classes.error_message}>{errors.overview}</div>
      )}
    </div>
  );
};

TextArea.propTypes = {
  title: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default TextArea;
