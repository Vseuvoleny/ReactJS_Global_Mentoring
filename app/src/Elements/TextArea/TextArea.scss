@import "../../styles/colors/colors.scss";

.textarea-container {
  margin-bottom: 60px;
  &__title {
    font-weight: 600;
    font-size: 16px;
    line-height: 20px;
    letter-spacing: 1px;
    text-transform: uppercase;
    color: $color-corral;
    opacity: 0.8;
  }
  &__textarea {
    margin-top: 13px;
    display: block;
    resize: none;
    outline: none;
    border: 0;
    background: rgba(50, 50, 50, 0.948044);
    mix-blend-mode: normal;
    opacity: 0.8;
    border-radius: 4px;
    padding: 18px;
    width: 100%;
    height: 197px;
    color: $color-white;
    font-size: 20px;
    line-height: 24px;
    caret-color: $color-white;
    &::placeholder {
      font-size: 20px;
      line-height: 24px;
      color: $color-white;
      opacity: 0.3;
    }
  }
  .error-message {
    padding-top: 4px;
    color: $color-corral;
  }
}
