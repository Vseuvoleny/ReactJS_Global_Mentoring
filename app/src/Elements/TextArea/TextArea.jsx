import React from "react";
import PropTypes from "prop-types";
import "./TextArea.scss";

const TextArea = ({ title, value, onChange }) => {
  return (
    <div className="textarea-container">
      <label html={title} className="textarea-container__title">
        {title}
      </label>
      <textarea
        id={title}
        className="textarea-container__textarea"
        placeholder="Movie description"
        value={value}
        onChange={onChange}
      ></textarea>
    </div>
  );
};

TextArea.propTypes = {
  title: PropTypes.string,
  value: PropTypes.string,
  onChange: PropTypes.func,
};

export default TextArea;
