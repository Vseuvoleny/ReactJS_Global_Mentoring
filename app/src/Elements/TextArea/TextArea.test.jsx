import React from "react";
import { render } from "@testing-library/react";
import TextArea from "./TextArea";

describe("TextArea component", () => {
  test("should render TextArea component", () => {
    const { container, getByPlaceholderText } = render(<TextArea />);

    const textArea = getByPlaceholderText("Movie description");
    const errorMessage = textArea.nextSibling;
    expect(errorMessage).not.toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
  test("should render TextArea component with error message", () => {
    const touched = { overview: "touched" };
    const { getByPlaceholderText } = render(
      <TextArea errors={"custom_error"} touched={touched} />
    );
    const textArea = getByPlaceholderText("Movie description");
    const errorMessage = textArea.nextSibling;
    expect(errorMessage).toBeInTheDocument();
  });
});
