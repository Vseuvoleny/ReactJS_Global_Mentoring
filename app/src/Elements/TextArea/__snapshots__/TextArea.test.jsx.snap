// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea component should render TextArea component 1`] = `
<div>
  <div
    class="textarea-container"
  >
    <label
      class="textarea-container__title"
    />
    <textarea
      class="textarea-container__textarea"
      placeholder="Movie description"
    />
  </div>
</div>
`;
